### PR TITLE
[Intermediate Representation] Generate the XgBoost training program

### LINKED ISSUE #827: [Intermediate Representation] Generate the XgBoost training program

### PR DESCRIPTION
**Pull request by tonyyang-svail**
Fix #827.

### CODE REVIEW

File: sql/codegen/xgboost/codegen.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	for _, attr := range attrs {
+		if _, ok := attrNames[attr.Key]; ok {
+			return nil, fmt.Errorf("duplicated attribute %s", attr.Key)
+		}
+		attrNames[attr.Key] = true
+		checker, ok := attributeChecker[attr.Key]
+		if !ok {
+			return nil, fmt.Errorf("unrecognized attribute %v", attr.Key)
+		}
+		if err := checker(attr.Value); err != nil {
```
Comment (typhoonzero) [inline sql/codegen/xgboost/codegen.go:84]: In the previous implementation, we intend to put all attributes without a prefix directly into `params` and then generate the configuration JSON string. Seems not quite the same here?

File: sql/codegen/xgboost/codegen.go [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+		}
+		return nil
+	},
+}
+
+func resolveModelType(estimator string) (string, error) {
+	switch strings.ToUpper(estimator) {
+	case "XGBOOST.GBTREE":
+		return "gbtree", nil
+	case "XGBOOST.GBLINEAR":
```
Comment (Yancey0623) [inline sql/codegen/xgboost/codegen.go:61]: For the booster algorithms except `gbtree`, we need to specify it in parameters:
``` python
param = {'booster': 'dart',
         'max_depth': 5, 'learning_rate': 0.1,
```
ref: https://xgboost.readthedocs.io/en/latest/tutorials/dart.html#sample-script

Maybe we can only support the `gbtree` booster and support other booster algs with the unit tests in the future?

Comment (tonyyang-svail) [reply sql/codegen/xgboost/codegen.go:61]: @Yancey1989 The current code does support `param = {'booster': 'dart'}` on line 128 `params["model."]["booster"] = booster`.

File: sql/codegen/xgboost/codegen.go (hunk 1 above)
Comment (tonyyang-svail) [reply sql/codegen/xgboost/codegen.go:84]: Hi @typhoonzero, which previous implementation are you referring to?

Comment (typhoonzero) [reply sql/codegen/xgboost/codegen.go:84]: In here: https://github.com/sql-machine-learning/sqlflow/blob/develop/sql/codegen_xgboost.go#L48

Comment (tonyyang-svail) [reply sql/codegen/xgboost/codegen.go:84]: I see. I will remove the `model.` prefix on the model parameters.

File: sql/codegen/xgboost/codegen.go (hunk 2 above)
Comment (Yancey0623) [reply sql/codegen/xgboost/codegen.go:61]: I see, thanks.

File: sql/codegen/xgboost/codegen.go (hunk 1 above)
Comment (tonyyang-svail) [reply sql/codegen/xgboost/codegen.go:84]: Done.

**Review by typhoonzero — APPROVED** [review]
LGTM, should add all XGBoost supported parameters in `attributeChecker`

**Pull request merged**